csharp/extractor/Semmle.Extraction.CIL/Entities/Method.cs

```diff
@@ -208,7 +208,7 @@ public override IEnumerable<IExtractionProduct> Contents
 
                 PopulateParameters(typeSignature.ParameterTypes);
 
-                foreach (var c in Parameters)
+                foreach (var c in Parameters!)
```

Comment:
Why not just produce no results if `Parameters` is `null`?

Reply:
`Parameters` is never null here, because it's accessed after `PopulateParameters()` so we can safely use the `!`. But adding `?? Array.Empty<Parameter>()` also works for me if that's what you meant?

Reply:
You can also use the `EnumerateNull` extension method from `Semmle.Util`.

Reply:
I just changed this to use `EnumerateNull`, but we're not winning anything because a couple of lines below we're indexing into the `Parameters` list, and I would need to add the `!` there to not get a warning.
So in the end, I'm not going to change this line, I'll leave it as `foreach (var c in Parameters!)`.

Reply:
Actually, I'm going to change `PopulateParameters` to return the parameters, and make the assignment to `Parameters` in the method, and then the compiler will know that it's assigned.

Reply:
👍